src/core/shape_inference/include/dimension_util.hpp

```diff
@@ -50,7 +50,8 @@ constexpr auto dilated(const TDim& dim, const typename TDim::value_type dilation
 template <class TDim>
 constexpr typename std::enable_if<std::is_arithmetic<TDim>::value, TDim>::type padded(const TDim dim,
                                                                                       const int64_t pad_num) {
-    return ((dim == inf_bound) || (dim + pad_num < 0)) ? inf_bound : dim + pad_num;
+    return ((static_cast<TDim>(dim) == static_cast<TDim>(inf_bound)) || (dim + pad_num < 0)) ? inf_bound
```

Comment:
dim is already TDim type, just cast to the `inf_bound` ?

Reply:
Done @zhangYiIntel